src/PKSim.Core/Services/PopulationSimulationBodyWeightUpdater.cs

```diff
@@ -0,0 +1,44 @@
+﻿using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Services;
+using PKSim.Core.Model;
+
+namespace PKSim.Core.Services
+{
+   public interface IPopulationSimulationBodyWeightUpdater
+   {
+      void UpdateBodyWeightForIndividual(PopulationSimulation populationSimulation, int individualId);
+      void ResetBodyWeightParameter(PopulationSimulation populationSimulation);
+   }
+
+   public class PopulationSimulationBodyWeightUpdater : IPopulationSimulationBodyWeightUpdater
+   {
+      private readonly IEntityPathResolver _entityPathResolver;
+
+      public PopulationSimulationBodyWeightUpdater(IEntityPathResolver entityPathResolver)
+      {
+         _entityPathResolver = entityPathResolver;
+      }
+
+      public void UpdateBodyWeightForIndividual(PopulationSimulation populationSimulation, int individualId)
```

Comment:
Is there a reason why this was moved out of the PK Stuff? Because you are doing line 24 to 26 now for EAch individual as opposed to one time.

Reply:
The only reason for it was that originally  I was reproducing the scaling logic in a second location. I think it makes sense to put it back now.